examples/tutorials/self-paced-training/part-4_advanced_federated_learning/chapter-10_federated_XGBoost/10.2_secure_fed_xgboost/secure_fed_xgboost.ipynb

```diff
@@ -0,0 +1,278 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "d8a97688-34ef-4425-9b25-d4c9fb086ad5",
+   "metadata": {},
+   "source": [
+    "# Secure Federated XGBoost with Homomorphic Encryption\n",
+    "This section illustrates the use of [NVIDIA FLARE](https://nvflare.readthedocs.io/en/main/index.html) enabling secure federated [XGBoost](https://github.com/dmlc/xgboost) under both horizontal and vertical collaborations.\n",
+    "The examples are based on a [finance dataset](https://www.kaggle.com/datasets/mlg-ulb/creditcardfraud) to perform fraud detection.\n",
+    "\n",
+    "## Secure Federated Training of XGBoost\n",
+    "In last section, we visited several mechanisms for training an XGBoost model in a federated learning setting, including histogram-based vertical, histogram-based horizontal, and tree-based horizontal methods. \n",
+    "\n",
+    "In this example, we further extend the existing histogram-based horizontal and vertical federated learning approaches to support secure federated learning using homomorphic encryption. Depending on the characteristics of the data to be encrypted, we can choose between [CKKS](https://github.com/OpenMined/TenSEAL) and [Paillier](https://github.com/intel/pailliercryptolib_python).\n",
+    "\n",
+    "In the following, we illustrate both *histogram-based* *horizontal* and *vertical* federated XGBoost, *with* homomorphic encryption. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "39f5d29c-0f4b-484e-99d7-75adf79a996c",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "Install required packages for data download and training"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "360512ae-bf5b-4868-893f-1e8df392bc47",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install -r requirements.txt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c66f4223-63a2-438c-97cd-c0bfbdd6c442",
+   "metadata": {},
+   "source": [
+    "## Encryption Plugins\n",
+    "The secure XGBoost requires encryption plugins to work. The plugins are distributed with NVFlare package. If you build NVFlare from source, you need\n",
```

Comment:
Do we distribute those ".so" now with pypi package? @IsaacYangSLA I forgot if we decide to include or NOT to include

Reply:
Thanks! Let me know if we have updated way of doing the pip install (I remember someone raised an issue for this?)